test_project/fastpg/db.py

```
@@ -84,34 +84,58 @@ async def fetch_all(self, query: str, values: Optional[Dict] = None):
 
-        transaction = await self.database.transaction()
+        if transaction is not None:
+            return await self.database.execute_many(query=query, values=list_of_values)
+
+        transaction_ctx = await self.database.transaction()
         try:
             result = await self.database.execute_many(query=query, values=list_of_values)
         except Exception as e:
-            await transaction.rollback()
+            await transaction_ctx.rollback()
             raise e
```

Ruff (0.14.7)

Use `raise` without specifying exception name

Remove exception name

(TRY201)